packages/databases-collections-list/src/items-table.tsx

```diff
@@ -0,0 +1,604 @@
+      button: {
+        opacity: 1,
+      },
+    },
+  },
+});
+
+// When row is not hovered, we hide the delete button
+const actionsCellStyles = css({
+  button: {
+    opacity: 0,
+    '&:focus': {
+      opacity: 1,
+    },
+  },
+  minWidth: spacing[800],
+});
+
+type ItemAction = 'delete';
+
+// Helper: Build actions array based on item state
+const buildItemActions = (
+  item: Item,
+  {
+    readOnly,
+    hasDeleteHandler,
+  }: { readOnly: boolean; hasDeleteHandler: boolean }
+): GroupedItemAction<ItemAction>[] => {
+  const actions: GroupedItemAction<ItemAction>[] = [];
+  if (!readOnly && hasDeleteHandler && !item.inferred_from_privileges) {
+    actions.push({
+      action: 'delete',
+      label: `Delete ${item.name}`,
+      tooltip: `Delete ${item.name}`,
+      icon: 'Trash',
+    });
+  }
+
+  return actions;
+};
+
+type ItemActionsProps = {
+  item: Item;
+  onDeleteItemClick?: (name: string) => void;
+};
+
+const ItemActions: React.FunctionComponent<ItemActionsProps> = ({
+  item,
+  onDeleteItemClick,
+}) => {
+  const { readOnly } = usePreferences(['readOnly']);
+  const itemActions = useMemo(
+    () =>
+      buildItemActions(item, {
+        readOnly,
+        hasDeleteHandler: !!onDeleteItemClick,
+      }),
+    [item, onDeleteItemClick, readOnly]
+  );
+
+  const onAction = useCallback(
+    (action: ItemAction) => {
+      if (action === 'delete') {
+        onDeleteItemClick?.(item._id);
+      }
+    },
+    [item, onDeleteItemClick]
+  );
+
+  return (
+    <ItemActionGroup<ItemAction>
+      data-testid="item-actions"
+      actions={itemActions}
+      onAction={onAction}
+    />
+  );
+};
+
+function calculateColumnsWithActions<T extends Item>(
+  columns: LGColumnDef<T>[],
+  onDeleteItemClick: ItemsTableProps<T>['onDeleteItemClick']
+) {
+  if (onDeleteItemClick) {
+    return [
+      ...columns,
+      {
+        id: 'actions',
+        header: '',
+        maxSize: 40,
+        cell: (info: CellContext<LGTableDataType<T>, unknown>) => {
+          return (
+            <ItemActions
+              item={info.row.original}
+              onDeleteItemClick={onDeleteItemClick}
+            />
+          );
+        },
+      },
+    ];
+  }
+  return columns;
+}
+
+type RowItem<T> = {
+  row: LeafyGreenTableRow<T>;
+  virtualRow?: LeafyGreenVirtualItem<T>;
+};
+
+const ItemsTableInner = <T extends Item>({
+  'data-testid': dataTestId,
+  namespace,
+  itemType,
+  items,
+  onItemClick,
+  onCreateItemClick,
+  onRefreshClick,
+  renderLoadSampleDataBanner,
+  tableContainerRef,
+  table,
+  rowItems,
+}: {
+  'data-testid'?: string;
+  namespace?: string;
+  itemType: 'collection' | 'database';
+  items: T[];
+  onItemClick: (id: string) => void;
+  onCreateItemClick?: () => void;
+  onRefreshClick?: () => void;
+  renderLoadSampleDataBanner?: () => React.ReactNode;
+  tableContainerRef?: React.RefObject<HTMLDivElement>;
+  // eslint-disable-next-line @typescript-eslint/no-redundant-type-constituents
```

Comment:
This is not needed (I'm guessing eslint was showing you the error after you added these new type exports, but the types weren't rebuilt)